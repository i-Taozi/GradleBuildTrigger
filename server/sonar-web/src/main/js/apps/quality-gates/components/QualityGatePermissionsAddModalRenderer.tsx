/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { ResetButtonLink, SubmitButton } from '../../../components/controls/buttons';
import Modal from '../../../components/controls/Modal';
import SelectLegacy from '../../../components/controls/SelectLegacy';
import GroupIcon from '../../../components/icons/GroupIcon';
import Avatar from '../../../components/ui/Avatar';
import { translate, translateWithParameters } from '../../../helpers/l10n';
import { Group, isUser } from '../../../types/quality-gates';

export interface QualityGatePermissionsAddModalRendererProps {
  onClose: () => void;
  onInputChange: (query: string) => void;
  onSubmit: (event: React.SyntheticEvent<HTMLFormElement>) => void;
  onSelection: (selection: T.UserBase | Group) => void;
  submitting: boolean;
  loading: boolean;
  query: string;
  searchResults: Array<T.UserBase | Group>;
  selection?: T.UserBase | Group;
}

type Option = (T.UserBase | Group) & { value: string };

export default function QualityGatePermissionsAddModalRenderer(
  props: QualityGatePermissionsAddModalRendererProps
) {
  const { loading, query = '', searchResults, selection, submitting } = props;

  const header = translate('quality_gates.permissions.grant');

  const noResultsText =
    query.length === 1 ? translateWithParameters('select2.tooShort', 2) : translate('no_results');

  return (
    <Modal contentLabel={header} onRequestClose={props.onClose}>
      <header className="modal-head">
        <h2>{header}</h2>
      </header>
      <form onSubmit={props.onSubmit}>
        <div className="modal-body">
          <div className="modal-field">
            <label>{translate('quality_gates.permissions.search')}</label>
            <SelectLegacy
              autoFocus={true}
              className="Select-big"
              clearable={false}
              // disable default react-select filtering
              filterOptions={i => i}
              isLoading={loading}
              noResultsText={noResultsText}
              onChange={props.onSelection}
              onInputChange={props.onInputChange}
              optionRenderer={optionRenderer}
              options={searchResults.map(r => ({ ...r, value: isUser(r) ? r.login : r.name }))}
              placeholder=""
              searchable={true}
              value={selection}
              valueRenderer={optionRenderer}
            />
          </div>
        </div>
        <footer className="modal-foot">
          {submitting && <i className="spinner spacer-right" />}
          <SubmitButton disabled={!selection || submitting}>{translate('add_verb')}</SubmitButton>
          <ResetButtonLink onClick={props.onClose}>{translate('cancel')}</ResetButtonLink>
        </footer>
      </form>
    </Modal>
  );
}

function optionRenderer(option: Option) {
  return (
    <>
      {isUser(option) ? (
        <Avatar hash={option.avatar} name={option.name} size={16} />
      ) : (
        <GroupIcon size={16} />
      )}
      <strong className="spacer-left">{option.name}</strong>
      {isUser(option) && <span className="note little-spacer-left">{option.login}</span>}
    </>
  );
}
