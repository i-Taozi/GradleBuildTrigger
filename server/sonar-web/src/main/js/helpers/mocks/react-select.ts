/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { GroupTypeBase, OptionProps } from 'react-select';

export function mockReactSelectOptionProps<
  OptionType,
  IsMulti extends boolean,
  GroupType extends GroupTypeBase<OptionType> = GroupTypeBase<OptionType>
>(
  data: OptionType,
  overrides?: OptionProps<OptionType, IsMulti, GroupType>
): OptionProps<OptionType, IsMulti, GroupType> {
  return {
    ...overrides,
    data
  } as OptionProps<OptionType, IsMulti, GroupType>;
}
