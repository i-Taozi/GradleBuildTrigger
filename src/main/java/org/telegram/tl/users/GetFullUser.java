/*
 *     This file is part of Telegram Server
 *     Copyright (C) 2015  Aykut Alparslan KOÃ‡
 *
 *     Telegram Server is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     Telegram Server is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.telegram.tl.users;

import org.telegram.core.TLContext;
import org.telegram.core.TLMethod;
import org.telegram.core.UserStore;
import org.telegram.data.UserModel;
import org.telegram.mtproto.ProtocolBuffer;
import org.telegram.tl.*;
import org.telegram.tl.contacts.*;
import org.telegram.tl.service.rpc_error;

public class GetFullUser extends TLObject implements TLMethod {

    public static final int ID = -902781519;

    public TLInputUser id;

    public GetFullUser() {
    }

    public GetFullUser(TLInputUser id){
        this.id = id;
    }

    @Override
    public void deserialize(ProtocolBuffer buffer) {
        id = (TLInputUser) buffer.readTLObject(APIContext.getInstance());
    }

    @Override
    public ProtocolBuffer serialize() {
        ProtocolBuffer buffer = new ProtocolBuffer(32);
        serializeTo(buffer);
        return buffer;
    }

    @Override
    public void serializeTo(ProtocolBuffer buff) {
        buff.writeInt(getConstructor());
        buff.writeTLObject(id);
    }

    public int getConstructor() {
        return ID;
    }

    @Override
    public TLObject execute(TLContext context, long messageId, long reqMessageId) {
        if (!context.isAuthorized()) {
            return new rpc_error(401, "UNAUTHORIZED");
        }
        int user_id = 0;
        if (this.id instanceof InputUserContact) {
            user_id = ((InputUserContact) this.id).user_id;
        } else if (this.id instanceof InputUserForeign) {
            user_id = ((InputUserForeign) this.id).user_id;
        } else if (this.id instanceof InputUserSelf) {
            UserModel um = UserStore.getInstance().getUser(context.getUserId());
            return new UserFull(um.toUser(context.getApiLayer()), new Link(new MyLinkContact(), new ForeignLinkMutual(), um.toUser(context.getApiLayer())),
                    new PhotoEmpty(), new PeerNotifySettingsEmpty(), false, new BotInfoEmpty());
        } else if (this.id instanceof InputUser) {
            user_id = ((InputUser) this.id).user_id;
        }
        UserModel um = UserStore.getInstance().getUser(user_id);
        if (um != null) {
            if (context.getApiLayer() >= 57) {
                org.telegram.tl.L57.UserFull userFull = new org.telegram.tl.L57.UserFull(0, um.toUser(context.getApiLayer()), "",
                        new org.telegram.tl.L57.contacts.Link(new org.telegram.tl.L57.ContactLinkContact(), new org.telegram.tl.L57.ContactLinkContact(), um.toUser(context.getApiLayer())),
                        null, new PeerNotifySettingsEmpty(), null);
                return userFull;
            } else {
                UserFull userFull = new UserFull(um.toUser(context.getApiLayer()), new Link(new MyLinkContact(), new ForeignLinkMutual(), um.toUser(context.getApiLayer())),
                        new PhotoEmpty(), new PeerNotifySettingsEmpty(), false, new BotInfoEmpty());
                return userFull;
            }

        }
        return null;
    }
}