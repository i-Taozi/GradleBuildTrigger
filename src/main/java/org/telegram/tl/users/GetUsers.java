/*
 *     This file is part of Telegram Server
 *     Copyright (C) 2015  Aykut Alparslan KOÃ‡
 *
 *     Telegram Server is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     Telegram Server is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.telegram.tl.users;

import org.telegram.core.TLContext;
import org.telegram.core.TLMethod;
import org.telegram.core.UserStore;
import org.telegram.data.UserModel;
import org.telegram.mtproto.ProtocolBuffer;
import org.telegram.tl.*;

public class GetUsers extends TLObject implements TLMethod {

    public static final int ID = 227648840;

    public TLVector<TLInputUser> id;

    public GetUsers() {
        this.id = new TLVector<>();
    }

    public GetUsers(TLVector<TLInputUser> id){
        this.id = id;
    }

    @Override
    public void deserialize(ProtocolBuffer buffer) {
        id = (TLVector<TLInputUser>) buffer.readTLObject(APIContext.getInstance());
    }

    @Override
    public ProtocolBuffer serialize() {
        ProtocolBuffer buffer = new ProtocolBuffer(32);
        serializeTo(buffer);
        return buffer;
    }

    @Override
    public void serializeTo(ProtocolBuffer buff) {
        buff.writeInt(getConstructor());
        buff.writeTLObject(id);
    }

    public int getConstructor() {
        return ID;
    }

    @Override
    public TLObject execute(TLContext context, long messageId, long reqMessageId) {
        TLVector<TLUser> users = new TLVector<>();
        for (TLInputUser u : this.id) {
            if (u instanceof InputUserContact) {
                UserModel um = UserStore.getInstance().getUser(((InputUserContact) u).user_id);
                users.add(um.toUser(context.getApiLayer()));
            } else if (u instanceof InputUserForeign) {
                UserModel um = UserStore.getInstance().getUser(((InputUserForeign) u).user_id);
                users.add(um.toUser(context.getApiLayer()));
            } else if (u instanceof InputUser) {
                UserModel um = UserStore.getInstance().getUser(((InputUser) u).user_id);
                users.add(um.toUser(context.getApiLayer()));
            }
        }

        return users;
    }
}