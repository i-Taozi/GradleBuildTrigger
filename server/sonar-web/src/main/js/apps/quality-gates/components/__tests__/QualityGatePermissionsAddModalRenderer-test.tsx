/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import SelectLegacy from '../../../../components/controls/SelectLegacy';
import { mockUserBase } from '../../../../helpers/mocks/users';
import QualityGatePermissionsAddModalRenderer, {
  QualityGatePermissionsAddModalRendererProps
} from '../QualityGatePermissionsAddModalRenderer';

it('should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot('default');
  expect(shallowRender({ query: 'a' })).toMatchSnapshot('short query');
  expect(shallowRender({ selection: mockUserBase() })).toMatchSnapshot('selection');
  expect(shallowRender({ selection: mockUserBase(), submitting: true })).toMatchSnapshot(
    'submitting'
  );
  expect(
    shallowRender({ query: 'ab', searchResults: [mockUserBase(), { name: 'group name' }] })
  ).toMatchSnapshot('query and results');
});

it('should render options correctly', () => {
  const wrapper = shallowRender();

  const { optionRenderer = () => null } = wrapper.find(SelectLegacy).props();

  expect(optionRenderer({ avatar: 'A', name: 'name', login: 'login' })).toMatchSnapshot('user');
  expect(optionRenderer({ name: 'group name' })).toMatchSnapshot('group');
});

function shallowRender(overrides: Partial<QualityGatePermissionsAddModalRendererProps> = {}) {
  return shallow(
    <QualityGatePermissionsAddModalRenderer
      loading={false}
      onClose={jest.fn()}
      onInputChange={jest.fn()}
      onSelection={jest.fn()}
      onSubmit={jest.fn()}
      query=""
      searchResults={[]}
      submitting={false}
      {...overrides}
    />
  );
}
