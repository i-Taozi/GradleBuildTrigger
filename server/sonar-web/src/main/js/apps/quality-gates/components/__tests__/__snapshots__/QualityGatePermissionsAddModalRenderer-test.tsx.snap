// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly: default 1`] = `
<Modal
  contentLabel="quality_gates.permissions.grant"
  onRequestClose={[MockFunction]}
>
  <header
    className="modal-head"
  >
    <h2>
      quality_gates.permissions.grant
    </h2>
  </header>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="modal-body"
    >
      <div
        className="modal-field"
      >
        <label>
          quality_gates.permissions.search
        </label>
        <SelectLegacy
          autoFocus={true}
          className="Select-big"
          clearable={false}
          filterOptions={[Function]}
          isLoading={false}
          noResultsText="no_results"
          onChange={[MockFunction]}
          onInputChange={[MockFunction]}
          optionRenderer={[Function]}
          options={Array []}
          placeholder=""
          searchable={true}
          valueRenderer={[Function]}
        />
      </div>
    </div>
    <footer
      className="modal-foot"
    >
      <SubmitButton
        disabled={true}
      >
        add_verb
      </SubmitButton>
      <ResetButtonLink
        onClick={[MockFunction]}
      >
        cancel
      </ResetButtonLink>
    </footer>
  </form>
</Modal>
`;

exports[`should render correctly: query and results 1`] = `
<Modal
  contentLabel="quality_gates.permissions.grant"
  onRequestClose={[MockFunction]}
>
  <header
    className="modal-head"
  >
    <h2>
      quality_gates.permissions.grant
    </h2>
  </header>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="modal-body"
    >
      <div
        className="modal-field"
      >
        <label>
          quality_gates.permissions.search
        </label>
        <SelectLegacy
          autoFocus={true}
          className="Select-big"
          clearable={false}
          filterOptions={[Function]}
          isLoading={false}
          noResultsText="no_results"
          onChange={[MockFunction]}
          onInputChange={[MockFunction]}
          optionRenderer={[Function]}
          options={
            Array [
              Object {
                "login": "userlogin",
                "value": "userlogin",
              },
              Object {
                "name": "group name",
                "value": "group name",
              },
            ]
          }
          placeholder=""
          searchable={true}
          valueRenderer={[Function]}
        />
      </div>
    </div>
    <footer
      className="modal-foot"
    >
      <SubmitButton
        disabled={true}
      >
        add_verb
      </SubmitButton>
      <ResetButtonLink
        onClick={[MockFunction]}
      >
        cancel
      </ResetButtonLink>
    </footer>
  </form>
</Modal>
`;

exports[`should render correctly: selection 1`] = `
<Modal
  contentLabel="quality_gates.permissions.grant"
  onRequestClose={[MockFunction]}
>
  <header
    className="modal-head"
  >
    <h2>
      quality_gates.permissions.grant
    </h2>
  </header>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="modal-body"
    >
      <div
        className="modal-field"
      >
        <label>
          quality_gates.permissions.search
        </label>
        <SelectLegacy
          autoFocus={true}
          className="Select-big"
          clearable={false}
          filterOptions={[Function]}
          isLoading={false}
          noResultsText="no_results"
          onChange={[MockFunction]}
          onInputChange={[MockFunction]}
          optionRenderer={[Function]}
          options={Array []}
          placeholder=""
          searchable={true}
          value={
            Object {
              "login": "userlogin",
            }
          }
          valueRenderer={[Function]}
        />
      </div>
    </div>
    <footer
      className="modal-foot"
    >
      <SubmitButton
        disabled={false}
      >
        add_verb
      </SubmitButton>
      <ResetButtonLink
        onClick={[MockFunction]}
      >
        cancel
      </ResetButtonLink>
    </footer>
  </form>
</Modal>
`;

exports[`should render correctly: short query 1`] = `
<Modal
  contentLabel="quality_gates.permissions.grant"
  onRequestClose={[MockFunction]}
>
  <header
    className="modal-head"
  >
    <h2>
      quality_gates.permissions.grant
    </h2>
  </header>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="modal-body"
    >
      <div
        className="modal-field"
      >
        <label>
          quality_gates.permissions.search
        </label>
        <SelectLegacy
          autoFocus={true}
          className="Select-big"
          clearable={false}
          filterOptions={[Function]}
          isLoading={false}
          noResultsText="select2.tooShort.2"
          onChange={[MockFunction]}
          onInputChange={[MockFunction]}
          optionRenderer={[Function]}
          options={Array []}
          placeholder=""
          searchable={true}
          valueRenderer={[Function]}
        />
      </div>
    </div>
    <footer
      className="modal-foot"
    >
      <SubmitButton
        disabled={true}
      >
        add_verb
      </SubmitButton>
      <ResetButtonLink
        onClick={[MockFunction]}
      >
        cancel
      </ResetButtonLink>
    </footer>
  </form>
</Modal>
`;

exports[`should render correctly: submitting 1`] = `
<Modal
  contentLabel="quality_gates.permissions.grant"
  onRequestClose={[MockFunction]}
>
  <header
    className="modal-head"
  >
    <h2>
      quality_gates.permissions.grant
    </h2>
  </header>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="modal-body"
    >
      <div
        className="modal-field"
      >
        <label>
          quality_gates.permissions.search
        </label>
        <SelectLegacy
          autoFocus={true}
          className="Select-big"
          clearable={false}
          filterOptions={[Function]}
          isLoading={false}
          noResultsText="no_results"
          onChange={[MockFunction]}
          onInputChange={[MockFunction]}
          optionRenderer={[Function]}
          options={Array []}
          placeholder=""
          searchable={true}
          value={
            Object {
              "login": "userlogin",
            }
          }
          valueRenderer={[Function]}
        />
      </div>
    </div>
    <footer
      className="modal-foot"
    >
      <i
        className="spinner spacer-right"
      />
      <SubmitButton
        disabled={true}
      >
        add_verb
      </SubmitButton>
      <ResetButtonLink
        onClick={[MockFunction]}
      >
        cancel
      </ResetButtonLink>
    </footer>
  </form>
</Modal>
`;

exports[`should render options correctly: group 1`] = `
<React.Fragment>
  <GroupIcon
    size={16}
  />
  <strong
    className="spacer-left"
  >
    group name
  </strong>
</React.Fragment>
`;

exports[`should render options correctly: user 1`] = `
<React.Fragment>
  <Connect(Avatar)
    hash="A"
    name="name"
    size={16}
  />
  <strong
    className="spacer-left"
  >
    name
  </strong>
  <span
    className="note little-spacer-left"
  >
    login
  </span>
</React.Fragment>
`;
